lib/config/configuration.go

```diff
@@ -686,6 +686,25 @@ func applyProxyConfig(fc *FileConfig, cfg *service.Config) error {
 		}
 		cfg.Proxy.TunnelPublicAddrs = addrs
 	}
+	if len(fc.Proxy.PostgresPublicAddr) != 0 {
+		// Postgres proxy is multiplexed on the web proxy port.
+		addrs, err := utils.AddrsFromStrings(fc.Proxy.PostgresPublicAddr, defaults.HTTPListenPort)
```

Comment:
what if this proxy is using a non-default web port?

Reply:
@awly Good question, I've actually spent quite some time thinking about what would the least confusing default behavior be in this case. I.e. consider config

```yaml
proxy_service:
  web_listen_addr: 0.0.0.0:8080
  public_addr: teleport.example.com:443
  postgres_public_addr: postgres.example.com
```

Which port would we default to in this case, 443, 8080, 3080? At first thought it seems logical to default to 443 but then we have this weird assumption that default port here depends on port in another field, and what if a user uses ALB, etc. So I ran through multiple scenarios like this in my head and in the end decided to keep it simple so we can just say "if unspecified, default port is 3080" which is very easy to reason about, both from UX and implementation standpoint. And then users can override it explicitly if needed (I almost think we should've made ports mandatory).

Maybe I'm overthinking this, idk, but to me this seemed to be the simplest way. Let me know if you think otherwise, I'm open to changing this.

Reply:
@awly After thinking about this in the back of my mind for a couple days, I think that defaulting to web proxy port would probably be the least surprising behavior after all. Updated.